### DISCUSSION TITLE
How to handle form errors an input when working with ant design

### DISCUSSION THREAD

**Discussion by davidshare** [react-hook-form/react-hook-form#3692] [General] [answered]
Please how do I display error messages for form elements - onChange?

https://codesandbox.io/s/romantic-cherry-2y24u?file=/src/App.js

I know I am doing something wrong, but I do not know what it is.

**Comment by bluebill1049** [comment, marked as answer]
here you go: https://codesandbox.io/s/damp-sound-dlyuc

couples things:

- `defaultValue` should be at Controller or at useForm with `defaultValues`
- `onChagne` from `Controller` should be wire up with your component
- `mode` should be changed to `onChange`

worth to spend some times with the doc, they are all documented.

**Reply by davidshare** [reply]
Thank you @bluebill1049 - actually feel embarrassed I did not pick those. Guess I was too eager to get it working and rushed the docs. Thank you.

**Reply by bluebill1049** [reply]
no worries at all @davidshare.
